### ISSUE TITLE
Missing FileInfo::set_attribute

### ISSUE THREAD

**Issue by GuillaumeGomez** [gtk-rs/gtk-rs#68]

**Comment by elmarco** [comment]
Does that look on the right path?
https://github.com/gtk-rs/gtk-rs/commit/bd74925488150fd2e980ef4e5f73078bdfa150f1

Unfortunately, we don't have ToGlibPtr for fundamental types. Am I missing something?

Otherwise, I think this function is not super useful, given that there are type-specific alternatives anyway. Could just be skipped imho.

**Comment by sdroege** [comment]
This needs to be manually bound around an enum that combines the `GFileAttributeType` and the `gpointer`. But I'm not sure what the point is, there are also `set_attribute_string()`, `set_attribute_uint64()`, etc. that cover all this too.

**Comment by sdroege** [comment]
I think this is supposed to take e.g. a `gboolean *` in case of `bool`, `guint32 *` in case of `u32`, etc. Your approach of implementing that will need something in this trait for getting a pointer.

**Comment by elmarco** [comment]
> I think this is supposed to take e.g. a `gboolean *` in case of `bool`, `guint32 *` in case of `u32`, etc. Your approach of implementing that will need something in this trait for getting a pointer.

Yes, I was expecting ToGlibPtr to be implemented for fundamental types too. Why not?

**Comment by sdroege** [comment]
Because that makes no sense generically. In most cases where you have a `guint32 *` it's actually an array where you somehow need to figure out the length, for example.
